corehq/apps/data_cleaning/utils/decorators.py

```diff
@@ -0,0 +1,20 @@
+import time
+from django.db import IntegrityError
+
+
+def retry_on_integrity_error(max_retries=3, delay=0.1):
```

Comment:
We already have a [`retry_on` decorator](https://github.com/dimagi/commcare-hq/blob/a9edcd7e3e75d3213ed08e2151fd6c42c25380f6/corehq/ex-submodules/dimagi/utils/retry.py) that does something similar to this, but with exponential backoff. You could use it to create a `retry_on_integrity_error` similar to this example:

https://github.com/dimagi/commcare-hq/blob/a9edcd7e3e75d3213ed08e2151fd6c42c25380f6/corehq/ex-submodules/dimagi/utils/couch/database.py#L173-L177